### PR TITLE
Bugfix/fix e2e tests response

### PR DESCRIPTION
**Pull request by JaviIG**
## Motivation and context

Types the response of the stubs used in the tests. This is to ensure that these test always have a valid response that fullfills the x-adapter interfaces

## Type of change
-  Bug fix (non-breaking change that fixes an issue)

## What is the destination branch of this PR?
-  `Main`

## How has this been tested?

Manually review of the test stubs.

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/tests/e2e/cucumber/spellcheck.spec.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
 import { And, Given, Then, When } from 'cypress-cucumber-preprocessor/steps';
+import { SearchResponse } from '@empathyco/x-adapter';
 
 Given('a results API response for a misspelled word', () => {
   cy.intercept('https://api.empathy.co/search', req => {
-    req.reply({
+    req.reply(<SearchResponse>{
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/spellcheck.spec.ts:6]: I think in that case we can remove all the response expects the spellcheck itself.

**Review by pmareke — APPROVED** [review]
For me is ok, the only thing is simplify the spellcheck response if you agree.

File: packages/x-components/tests/e2e/cucumber/search-box/search-box.spec.ts [hunk 2]
```diff
@@ -157,7 +158,7 @@ And('related tags are displayed after instantDebounceInMs is {boolean}', (instan
 // Scenario 4
 Given('a second results API with a known response', () => {
   cy.intercept('https://api.empathy.co/search', req => {
-    req.reply({
+    req.reply(<SearchResponse>{
```
Comment (mavmaf) [inline packages/x-components/tests/e2e/cucumber/search-box/search-box.spec.ts:161]: Uhm, is it working for you? I have to add the missing parts in the response so the editor doesn't 'yell'
```
req.reply(<SearchResponse>{
      banners: [],
      promoteds: [],
      facets: [],
      results: [
        createResultStub('LEGO Duplo Disney Tren de Cumpleaños de Mickey y Minnie - 10941', {
          images: ['https://picsum.photos/seed/8/100/100']
        }),
        createResultStub('LEGO Disney Granja de Mickey Mouse y el Pato Donald - 10775', {
          images: ['https://picsum.photos/seed/10/100/100']
        })
      ],
      redirections: [],
      partialResults: [],
      totalResults: 7,
      queryTagging: {
        url: 'https://tagging.empathy.co',
        params: {}
      },
      spellcheck: 'lego'
    });
```

File: packages/x-components/tests/e2e/cucumber/base-result-image.spec.ts [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
 
 Given('a results API with broken images', () => {
   cy.intercept('https://api.empathy.co/search', req => {
-    req.reply({
+    req.reply(<SearchResponse>{
       banners: [],
       promoteds: [],
+      redirections: [],
       facets: [],
+      totalResults: 3,
```
Comment (mavmaf) [inline packages/x-components/tests/e2e/cucumber/base-result-image.spec.ts:12]: `totalResults: 3` is duplicated. Please remove one of them

**Review by mavmaf — CHANGES_REQUESTED** [review]
Github says I need to add something here to request changes :D

File: packages/x-components/tests/e2e/cucumber/spellcheck.spec.ts (hunk 1 above)
Comment (JaviIG) [reply packages/x-components/tests/e2e/cucumber/spellcheck.spec.ts:6]: The `SearchResponse` interface says that every property is required. That´s what I´m doing in this PR, ensuring that these interfaces are fulfilled. A little bird has told me that the great @javieri-empathy wants to relax this interfaces, and make every property optional. After that relaxation I agree with your idea makes sense and can be done. However not right now.

File: packages/x-components/tests/e2e/cucumber/search-box/search-box.spec.ts (hunk 2 above)
Comment (JaviIG) [reply packages/x-components/tests/e2e/cucumber/search-box/search-box.spec.ts:161]: You are right, I forgot about this one.

File: packages/x-components/tests/e2e/cucumber/spellcheck.spec.ts (hunk 1 above)
Comment (pmareke) [reply packages/x-components/tests/e2e/cucumber/spellcheck.spec.ts:6]: What I meant is that you're mocking a huge `SearchResponse` when you only need the spellcheck in the spellcheck tests.

Maybe we can use a `Partial` here? I don't know man but for me this reply has more note than info.

Whatever.

**Review by pmareke — APPROVED** [review]

**Review by mavmaf — APPROVED** [review]

**Pull request merged**